lib/spack/spack/build_systems/autotools.py

```diff
@@ -151,6 +145,8 @@ def _do_patch_config_guess(self):
         if config_guess is not None:
             try:
                 check_call([config_guess], stdout=PIPE, stderr=PIPE)
+                mod = stat(my_config_guess).st_mode & 0777 | S_IWUSR
```

Comment:
oh wait -- maybe I misread this.  It's doing that already -- `0777` is just a mask on `st_mode`.